react-notes/src/layouts/MainLayout/components/Sidebar/components/ArticleLink/ArticleLink.js

```diff
@@ -1,22 +1,84 @@
-import { Link } from 'react-router-dom';
+import { useContext } from 'react';
+import { Link, useNavigate, useParams } from 'react-router-dom';
 import classNames from 'classnames';
 
+import { ArticlesContext } from '../../../../../../pages/article/context';
+
 import styles from './ArticleLink.module.css';
 
-export const ArticleLink = ({ articleData }) => (
-    <li className={styles.container}>
-        <Link to={`article/${articleData.id}`}>
-            <div
-                className={classNames(
-                    articleData.title ? styles.title : styles.untitled,
-                )}
-            >
-                {articleData.title || 'Untitled'}
-            </div>
-
-            <div className={styles.lessonDate}>
-                {articleData.lessonDate || 'no date'}
-            </div>
-        </Link>
-    </li>
-);
+export const ArticleLink = ({ articleData }) => {
+    const navigate = useNavigate();
+    const { fetchArticles, sendArticles } = useContext(ArticlesContext);
+    const { articleId: currentArticleId } = useParams();
+    const articleURL = `/article/${articleData.id}`;
+
+    const onDelete = () => {
+        const articles = fetchArticles();
+        sendArticles(
+            articles.filter((article) => article.id !== articleData.id),
+        );
+
+        if (articleData.id === currentArticleId) navigate('/');
+    };
+
+    const onEdit = () => {
+        navigate(`${articleURL}/edit`);
+    };
+
+    const linkOnClickHandle = (e) => {
+        if (e.target.tagName === 'svg') e.preventDefault();
+    };
+
+    return (
+        <li className={styles.container}>
+            <Link onClick={linkOnClickHandle} to={articleURL}>
+                <div>
+                    <div
+                        className={classNames(
+                            articleData.title ? styles.title : styles.untitled,
+                        )}
+                    >
+                        {articleData.title || 'Untitled'}
+                    </div>
+
+                    <div className={styles.lessonDate}>
+                        {articleData.lessonDate || 'no date'}
+                    </div>
+                </div>
+                <div className={styles.btnContainer}>
+                    <button className={styles.button} onClick={onEdit}>
+                        <svg
```

Comment:
I used svg tag because I need to manipulate with color with CSS, but I think it's problem that the image is not in seperate file but in HTML